### PR TITLE
Add testing infrastructure

### LINKED ISSUE #3: Figure out Testing Infrastructure
Find some sort of way to run tests, and ensure that things actually work! 

Maybe create an in-memory MongoDB (as per @anandani4136 's suggestion and go from there)? Maybe Jest?

### PR DESCRIPTION
**Pull request by Timothy-Gonzalez**
Resolves #3

This is a big PR. Here's what it does:
- [x] Rewrites how models are defined so that they can be initialized upon request instead of on import, allowing for easier testing.
- [x] Extracts the base url logic into it's own module for easier mocking. In the future, we should have all env variables defined in some module so it's not as awkward to mock them. (Created #90 for this)
- [x] Adds jest & a few other dependencies to allow for tests to be written.
- [x] Provides a basic sanity test for the app as well as for one of the user endpoints. More tests will need to written, but this at least lays the foundation.
- [x] Adds a github job to CI workflow to run said tests.

This took a lot of work because of how models were defined, going forward this should be much easier to work with.

Also, tests run in parallel because it's more future proof.

### CODE REVIEW

File: jest.config.ts
Comment (AydanPirani) [inline jest.config.ts:1]: Is it worth it to have all these config comments in here?

File: src/testTools.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import request from "supertest";
+
+import { Provider, Role } from "./services/auth/auth-models.js";
+
+export const TESTER = {
+    id: "bob-the-tester101010101011",
+    email: "bob-the-tester@hackillinois.org",
+    name: "Bob Tester",
```
Comment (AydanPirani) [inline src/testTools.ts:11]: what does this TESTER represent?

File: src/testTools.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    [Role.APPLICANT]: [Role.APPLICANT],
+    [Role.USER]: [Role.USER],
+    [Role.ATTENDEE]: [Role.ATTENDEE],
+    [Role.STAFF]: [Role.ATTENDEE, Role.STAFF],
+    [Role.ADMIN]: [Role.ATTENDEE, Role.STAFF, Role.ADMIN],
+
+    [Role.MENTOR]: [Role.MENTOR],
+    [Role.SPONSOR]: [Role.SPONSOR],
+    [Role.BLOBSTORE]: [Role.BLOBSTORE],
+};
```
Comment (AydanPirani) [inline src/testTools.ts]: we might need to make a few changes to this, can you add roles as such (note that we can never remove them):

image

**Review by AydanPirani — APPROVED** [review]
LGTM - please just make the user mapping changes, and then feel free to merge!

File: jest.config.ts
Comment (Timothy-Gonzalez) [reply jest.config.ts:1]: Resolved by acd1a43

File: src/testTools.ts (hunk 1 above)
Comment (Timothy-Gonzalez) [reply src/testTools.ts:11]: Resolved by beda79a

File: src/testTools.ts (hunk 2 above)
Comment (Timothy-Gonzalez) [reply src/testTools.ts]: Resolved by 17bacb6

**Pull request merged**